### PR TITLE
test: add retries to expected successful curl calls

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

**Comment by nebril** [comment]
test-4.9

**Comment by nebril** [comment]
test-4.9

**Comment by nebril** [comment]
test-4.9

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -949,15 +949,15 @@ var _ = Describe("K8sPolicyTest", func() {
 						namespaceForTest,
 						"vagrant-cache.ci.cilium.io.",
 					)
-					curlCmd = helpers.CurlFail(resource)
+					curlCmd = helpers.CurlFail("--retry 5 " + resource)
```
Comment (tgraf) [inline test/k8sT/Policies.go]: I'm not sure I really like this. Most of these are for internal destinations. We shouldn't depend on retries there. Can you pass in a flag instead which indicates whether to retry or not? We should only retry for the `worldTarget` case.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: test/k8sT/Policies.go (hunk 1 above)
Comment (nebril) [reply test/k8sT/Policies.go]: Makes sense, thanks!

**Comment by nebril** [comment]
test-focus K8sPolicyTest

**Review by tgraf — APPROVED** [review]

**Review by qmonnet — APPROVED** [review]

**Pull request merged**